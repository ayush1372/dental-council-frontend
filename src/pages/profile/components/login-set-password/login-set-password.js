import { useState } from 'react';

import InfoOutlinedIcon from '@mui/icons-material/InfoOutlined';
import TaskAltOutlinedIcon from '@mui/icons-material/TaskAltOutlined';
import { Box, Dialog, DialogActions, DialogContent, Typography } from '@mui/material';
import { useForm } from 'react-hook-form';

import { verboseLog } from '../../../../config/debug';
import { Button, RadioGroup, TextField } from '../../../../ui/core';
import { PasswordRegexValidation } from '../../../../utilities/common-validations';
import { ModalOTP } from '../../../login/components/modal-otp/modal-otp';

export function LoginSetPassword() {
  const { otpPopup, handleClickOpen, isOtpVerified } = ModalOTP();
  const handleChange = (event) => {
    setValue(event.target.name, event.target.value, true);
    handleClickOpen();
    verboseLog(isOtpVerified);
  };
  const [isDialogOpen, setIsDialogOpen] = useState(false);

  const {
    register,
    handleSubmit,
    setValue,
    getValues,
    watch,
    formState: { errors },
  } = useForm({
    mode: 'onChange',
    defaultValues: {
      SetPassword: '',
      ConfirmPassword: '',
    },
  });

  const handleDialogClose = () => {
    setIsDialogOpen(false);
  };

  const onSubmit = () => {
    setIsDialogOpen(true);
  };

  return (
    <Box mt={8}>
      <Typography variant="h1" mb={2}>
        Set Password
      </Typography>
      <Typography variant="body1">Here you can set password for your Account.</Typography>
      <Box
        border="1px solid "
        borderColor="inputBorderColor.main"
        borderRadius="3px"
        mt={8}
        p="24px 48px"
      >
        <Box mt={2}>
          <TextField
            fullWidth={true}
            data-testid="passwordtxt"
            variant="outlined"
            label=" Set Password"
            type="Password"
            name="SetPassword"
            required="true"
            placeholder="Set Password"
            margin="dense"
            defaultValue={getValues().SetPassword}
            error={errors.SetPassword?.message}
            {...register('SetPassword', PasswordRegexValidation)}
          />
        </Box>
        <Typography variant="caption" color="messageBlue.main">
          <InfoOutlinedIcon
            sx={{ fontSize: '15px', verticalAlign: 'middle', marginRight: '5px' }}
          />
          Your Password should contains Numbers(0-9)/Letters(a-z) and special characters.
        </Typography>
        <Box mt={2}>
          <TextField
            fullWidth={true}
            data-testid="confirmpasswordtxt"
            variant="outlined"
            type="Password"
            label="Confirm Password"
            name="ConfirmPassword"
            required="true"
            placeholder="Confirm Password"
            margin="dense"
            defaultValue={getValues().ConfirmPassword}
            error={errors.ConfirmPassword?.message}
            {...register('ConfirmPassword', {
              required: true,
              validate: (val = '') => {
                if (watch('SetPassword') !== val) {
                  return 'Your passwords do no match';
                }
              },
            })}
          />
        </Box>
        <Box>
          <Typography variant="body1" sx={{ mt: 4 }} component="div">
            Validate Using
          </Typography>
          <Box
            borderRadius="3px"
            border="1px solid"
            borderColor="inputBorderColor.main"
            p="14px 20px"
            mb="25px"
          >
            <RadioGroup
              value={getValues().verifyOption}
              onChange={handleChange}
              name={'verifyOption'}
              data-testid="linkguardian"
              disabled={isOtpVerified}
              items={[
                {
                  value: 'OTP on Mobile number linked with Aadhaar',
                  label: 'OTP on Mobile number linked with Aadhaar',
                },
                {
                  value: 'OTP on Mobile number linked with ABHA Number',
                  label: 'OTP on Mobile number linked with ABHA Number',
                },
              ]}
            />
          </Box>
          {otpPopup}
          {isOtpVerified}
          <Box display="flex" justifyContent="flex-end" width="100%">
            <Button
              data-testid="save"
              color="secondary"
              variant="contained"
              disabled={!isOtpVerified}
              onClick={handleSubmit(onSubmit)}
            >
              Submit
            </Button>
          </Box>
          <Dialog
            open={isDialogOpen}
            onClose={handleDialogClose}
            aria-labelledby="alert-dialog-title"
            aria-describedby="alert-dialog-description"
          >
            <DialogContent
              sx={{
                width: '500px',
                padding: '0px',
              }}
            >
              <Box
                sx={{
                  marginTop: '48px',
                }}
              >
                <Box
                  sx={{
                    display: 'flex',
                    justifyContent: 'center',
                  }}
                >
                  <TaskAltOutlinedIcon
                    sx={{
                      color: 'success.dark',
                      width: '80px',
                      height: '80px',
                    }}
                  />
                </Box>
                <Box
                  sx={{
                    textAlign: 'center',
                  }}
                >
                  <Typography variant="h1" color="success.dark">
                    Success!
                  </Typography>
                </Box>
                <Box
                  sx={{
                    textAlign: 'center',
                  }}
                >
                  <Typography color="grey.context" variant="subtitle1">
                    Your Password has been <br /> successfully created.
                  </Typography>
                </Box>
              </Box>
            </DialogContent>
            <DialogActions
              sx={{
                padding: '32px 48px 48px 48px',
              }}
            >
              <Button
                variant="contained"
                color="secondary"
                fullWidth
                onClick={handleDialogClose}
                autoFocus
              >
                View My Profile
              </Button>
            </DialogActions>
          </Dialog>
        </Box>
      </Box>
    </Box>
  );
}

export default LoginSetPassword;
