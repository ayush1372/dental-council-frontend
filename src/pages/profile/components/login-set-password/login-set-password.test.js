import '@testing-library/jest-dom';

import { render, screen } from '@testing-library/react';
import { fireEvent } from '@testing-library/react';

import { LoginSetPassword } from './login-set-password';

test('Set Password component snapshot', () => {
  const { container } = render(<LoginSetPassword />);
  expect(container).toMatchSnapshot();
});

it('should render successfully', async () => {
  render(<LoginSetPassword />);
  expect(screen.getByText('OTP on Mobile number linked with Aadhaar')).toBeInTheDocument();
  expect(screen.getByText('OTP on Mobile number linked with ABHA Number')).toBeInTheDocument();
});

it('should check Radio Button to be checked ', () => {
  render(<LoginSetPassword />);

  const firstRadio = screen.getByLabelText('OTP on Mobile number linked with Aadhaar');
  fireEvent.click(firstRadio);
  expect(firstRadio).toBeChecked();

  const secondRadio = screen.getByLabelText('OTP on Mobile number linked with ABHA Number');
  fireEvent.click(secondRadio);
  expect(secondRadio).toBeChecked();
});

test('Renders Set Password Field', () => {
  render(<LoginSetPassword />);
  const TextElement = screen.getByTestId('passwordtxt');
  fireEvent.click(TextElement);
  const PasswordText = screen.getByText('Set Password');
  expect(PasswordText).toBeInTheDocument();
});

test('Renders Confirm Password Field', () => {
  render(<LoginSetPassword />);
  const TextElement = screen.getByTestId('confirmpasswordtxt');
  fireEvent.click(TextElement);
  const ConfirmPasswordText = screen.getByText('Confirm Password*');
  expect(ConfirmPasswordText).toBeInTheDocument();
});
