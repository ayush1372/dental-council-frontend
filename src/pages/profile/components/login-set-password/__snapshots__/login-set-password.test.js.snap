// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Set Password component snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-h8m1q6"
  >
    <h1
      class="MuiTypography-root MuiTypography-h1 css-l4yn1p-MuiTypography-root"
    >
      Set Password
    </h1>
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      Here you can set password for your Account.
    </p>
    <div
      class="MuiBox-root css-bjfkvu"
    >
      <div
        class="MuiBox-root css-1yuhvjn"
      >
        <div
          class="MuiBox-root css-0"
        >
          <span
            class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          >
             Set Password
          </span>
          <p
            class="MuiTypography-root MuiTypography-body2 css-uhz803-MuiTypography-root"
          >
             *
          </p>
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
          data-testid="passwordtxt"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              autocomplete="off"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
              id="mui-1"
              name="SetPassword"
              placeholder="Set Password"
              type="Password"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
            >
              <button
                aria-label="toggle password visibility"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="VisibilityIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-ihdtdm"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <span
        class="MuiTypography-root MuiTypography-caption css-z021j0-MuiTypography-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1mekref-MuiSvgIcon-root"
          data-testid="InfoOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 7h2v2h-2zm0 4h2v6h-2zm1-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
          />
        </svg>
        Your Password should contains Numbers(0-9)/Letters(a-z) and special characters.
      </span>
      <div
        class="MuiBox-root css-1yuhvjn"
      >
        <div
          class="MuiBox-root css-0"
        >
          <span
            class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          >
            Confirm Password*
          </span>
          <p
            class="MuiTypography-root MuiTypography-body2 css-uhz803-MuiTypography-root"
          >
             *
          </p>
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
          data-testid="confirmpasswordtxt"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              autocomplete="off"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
              id="mui-2"
              name="ConfirmPassword"
              placeholder="Confirm Password"
              type="Password"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
            >
              <button
                aria-label="toggle password visibility"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="VisibilityIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-ihdtdm"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="MuiTypography-root MuiTypography-body1 css-rcjyty-MuiTypography-root"
        >
          Validate Using
        </div>
        <div
          class="MuiBox-root css-1avrwyx"
        >
          <fieldset
            class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
          >
            <div
              class="MuiFormGroup-root css-18nkkfo-MuiFormGroup-root"
              role="radiogroup"
            >
              <label
                class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                data-testid="linkguardian"
              >
                <span
                  class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary css-vqmohf-MuiButtonBase-root-MuiRadio-root"
                >
                  <input
                    class="PrivateSwitchBase-input css-1m9pwf3"
                    name="verifyOption"
                    type="radio"
                    value="OTP on Mobile number linked with Aadhaar"
                  />
                  <span
                    class="css-hyxlzm"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                      data-testid="RadioButtonUncheckedIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                      data-testid="RadioButtonCheckedIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </span>
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                >
                  OTP on Mobile number linked with Aadhaar
                </span>
              </label>
              <label
                class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                data-testid="linkguardian"
              >
                <span
                  class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary css-vqmohf-MuiButtonBase-root-MuiRadio-root"
                >
                  <input
                    class="PrivateSwitchBase-input css-1m9pwf3"
                    name="verifyOption"
                    type="radio"
                    value="OTP on Mobile number linked with ABHA Number"
                  />
                  <span
                    class="css-hyxlzm"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                      data-testid="RadioButtonUncheckedIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                      data-testid="RadioButtonCheckedIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </span>
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                >
                  OTP on Mobile number linked with ABHA Number
                </span>
              </label>
            </div>
          </fieldset>
        </div>
        <div
          class="MuiBox-root css-0"
        />
        <div
          class="MuiBox-root css-asvefe"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-zcbmsk-MuiButtonBase-root-MuiButton-root"
            data-testid="save"
            tabindex="0"
            type="button"
          >
            Submit
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
